share/getters/cascade.go

```diff
@@ -125,7 +124,7 @@ func cascadeGetters[V any](
 		if getErr == nil {
 			return val, nil
 		}
-		if errors.Is(share.ErrNamespaceNotFound, getErr) {
+		if errors.Is(getErr, share.ErrNamespaceNotFound) || errors.Is(getErr, context.Canceled) {
```

Comment:
Do we want to track user deadline here?

Reply:
Good find! Yeah, we also want to keep previous errors. I've reordered err handlers